File: code/Python/Strings/Day_14_Atoi.py
class Solution:
    def myAtoi(self, s: str) -> int:
        i=0
        n=len(s)
        sign=1
        res=0
        
        while i<n and s[i]==' ':
            i+=1
        
        if i<n and (s[i]=='-' or s[i]=='+'):
            if s[i]=='-':
                sign = -1
            i+=1
        
        while i<n and '0'<=s[i]<='9':
            res = res*10 + (ord(s[i])-ord('0'))
            i+=1
        
        res*=sign
        
        maxlim = 2**31 - 1
        minlim = -2**31
        
        if res > maxlim:
            return maxlim
        if res < minlim:
            return minlim
        
        return res

File: code/Cpp/Strings/Day_14_Atoi.cpp
class Solution {
  public:
    int myAtoi(char *s) {
        
        int sign=1;
        long int res=0;
        int n=strlen(s);
        int i=0;
        
        while(i<n && s[i]==' '){
            i++;
        }
        
        while(i<n && (s[i]=='-'||s[i]=='+')){
            if(s[i]=='-'){
                sign=-1;
            }
            i++;
        }
        
        while(i<n && '0'<=s[i] && s[i]<='9'){
            res = res*10 + (s[i] - '0');
            i++;
        }
        res*=sign;
        
        if(res>INT_MAX) res=INT_MAX;
        else if(res<INT_MIN) res=INT_MIN;
        
        return res;
        
    }
};